### ISSUE TITLE
Add alpha-value-notation

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#4720]
> What is the problem you're trying to solve?

I'd like to enforce a consistent alpha-value notation.

From https://github.com/stylelint/stylelint/issues/4718#issuecomment-619336037:

"consider being opinionated w.r.t. how to represent alpha values. IMHO using percentage notation is most readable, as it clarifies the range: e.g. `70%` vs `0.7`"

References:

- [Spec](https://www.w3.org/TR/css-color-4/#typedef-alpha-value)
- [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/alpha-value)

> What solution would you like to see?

A new rule that lets me enforce either a `<number>` or `<percentage>` notation for `<alpha-value>`.

Primary option: `string: "number"|"percentage"`

With `"number"` the following are valid:

```css
a { 
  color: rgba(0, 0, 0, 0.5)
  color: rgb(0 0 0 / 0.5)
  color: hsl(198deg 28% 50% / 0.5);
  opacity: 0.5;
  shape-image-threshold: 0.5;
  ..
}
```

With `"percentage"` the following are valid:

```css
a { 
  color: rgba(0, 0, 0, 50%)
  color: rgb(0 0 0 / 50%)
  color: hsl(198deg 28% 50% / 50%);
  opacity: 50%;
  shape-image-threshold: 50%;
  ..
}
```

Any thoughts about the use case, name or options?

**Comment by jeddy3** [comment]
As there are no objections to this rule, I'll label it as good-to-go.

- Name: `alpha-value-notation`
- Primary option: `"percentage"|"number"`
- Secondary options: None
- Message: `Expected "${unfixed}" to be "${fixed}"`
- Description: "Specify percentage or number notation for alpha-values."
- Section: ["Limit language features" -> "Color"](https://stylelint.io/user-guide/rules/list#color-1)

Please comment below if you'd like to [contribute](https://stylelint.io/contributing) this rule to stylelint.

There are [steps on how to add a new rule](https://stylelint.io/developer-guide/rules#add-a-rule) in the Developer guide.
